homeassistant/components/fan/dyson.py

```diff
@@ -238,6 +288,210 @@ def supported_features(self) -> int:
     def device_state_attributes(self) -> dict:
         """Return optional state attributes."""
         return {
-            ATTR_IS_NIGHT_MODE: self.is_night_mode,
-            ATTR_IS_AUTO_MODE: self.is_auto_mode
+            ATTR_NIGHT_MODE: self.night_mode,
+            ATTR_AUTO_MODE: self.auto_mode
             }
+
+
+class DysonPureCoolDevice(FanEntity):
+    """Representation of a Dyson Purecool (TP04/DP04) fan."""
+
+    def __init__(self, hass, device):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.